### ISSUE TITLE
Expensive operations

### ISSUE THREAD

**Issue by olamiko** [yhabib/csi#2]
In the [line](https://github.com/yhabib/csi/blob/658526d5315ec219fcd95c0c04bdc01ddd80fc35/prework/Go/assignment/xkdc.go#L108), you get comics but return the path to file. In the `searchForPattern` function, you perform a read on the file on this [line](https://github.com/yhabib/csi/blob/658526d5315ec219fcd95c0c04bdc01ddd80fc35/prework/Go/assignment/xkdc.go#L80). Those are duplicate operations that can be done only once if you pass the `comics` variable from `createIndex` to the `searchForPattern` function. What do you think? I would reduce expensive operations

**Comment by yhabib** [comment]
That is true 🤔  I thought this approach would make the code cleaner, as the path becomes the "resource" and `main` doesn't need to know much about it. But from a performance point of view your point makes sense, if `main` handles `comics` one would fill it and the other one would consume, but both working with the pointer.
